src/transformers/adapters/peft_mixin.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from typing import Optional
+
+from ..utils import ADAPTER_CONFIG_NAME, cached_file, logging, requires_backends
+
+
+logger = logging.get_logger(__name__)
+
+
+class PeftAdapterMixin:
+    """
+    A class containing all functions for loading and using adapters weights that are supported in PEFT library.
+    Currently supported PEFT methods are all non-prefix tuning methods
+    """
+
+    def load_adapter(
+        self,
+        peft_model_id: str,
+        adapter_name: Optional[str] = "default",
+        revision: Optional[str] = None,
+        use_auth_token: Optional[str] = None,
+        commit_hash: Optional[str] = None,
+    ):
+        """
+        Load adapter weights from file. Requires peft as a backend to load the adapter weights
+        """
+        requires_backends(self.load_adapter, "peft")
+
+        from peft import LoraConfig, PeftModel, create_and_replace
+        from peft.utils import set_peft_model_state_dict
+        from peft.utils.other import TRANSFORMERS_MODELS_TO_LORA_TARGET_MODULES_MAPPING
+
+        self.peft_config = {}
```

Comment:
I would suggest to create this attribute only after the final check (`if adapter_name not in self.peft_config`) is done. So this line would become `peft_config = {}`, and then after the mentioned check, `self.peft_config = peft_config`. That way, if something goes wrong and an error is raised, the attribute is not created needlessly on the model.

Reply:
Sounds great! Thanks for the suggestion, will keep this unresolved until I will propose something

Reply:
This is not needed anymore, please check my comment above as well as this specific commit: https://github.com/huggingface/peft/pull/749/commits/8053c88bbdd1651cd75b59b6cb0111cdb1ddd41b